hooks/tk-multi-publish2/basic/publish_vred.py

```diff
@@ -0,0 +1,763 @@
+        when multiple tasks have been selected.
+
+        Example::
+
+            {
+                 "setting_a": "/path/to/a/file"
+            }
+
+        :param widget: The widget that was created by `create_settings_widget`
+        :param items:  A list of PublishItems the selected publish tasks are parented to
+        """
+
+        ui_settings = {
+            "context": widget.context,
+            "filename": widget.filename.text(),
+            "publish_to_sg": widget.publish_to_shotgrid.isChecked(),
+        }
+
+        return {"UI Settings": ui_settings}
+
+    def set_ui_settings(self, widget, settings, items=None):
+        """
+        Allows the custom UI to populate its fields with the settings from the
+        currently selected tasks.
+
+        The widget is the widget created and returned by
+        `create_settings_widget`.
+
+        A list of settings dictionaries are supplied representing the current
+        values of the settings for selected tasks. The settings dictionaries
+        correspond to the dictionaries returned by the settings property of the
+        hook.
+
+        Example::
+
+            settings = [
+            {
+                 "setting_a": "/path/to/a/file"
+                 "setting_b": False
+            },
+            {
+                 "setting_a": "/path/to/a/file"
+                 "setting_b": False
+            }]
+
+        The default values for the settings will be the ones specified in the
+        environment file. Each task has its own copy of the settings.
+
+        When invoked with multiple settings dictionaries, it is the
+        responsibility of the custom UI to decide how to display the
+        information. If you do not wish to implement the editing of multiple
+        tasks at the same time, you can raise a ``NotImplementedError`` when
+        there is more than one item in the list and the publisher will inform
+        the user than only one task of that type can be edited at a time.
+
+        :param widget: The widget that was created by `create_settings_widget`
+        :param settings: a list of dictionaries of settings for each selected
+            task.
+        :param items:  A list of PublishItems the selected publish tasks are parented to
+        """
+
+        if len(settings) > 1:
+            raise NotImplementedError()
+
+        # initialize description widget
+        widget.description_label.setText(self.description)
+
+        # restore previous saved UI settings
+        ui_settings = settings[0]["UI Settings"]
+
+        # if no Publish template has been defined in the plugin setting, hide the publish option from the UI
+        publish_template = self.get_publish_template(settings[0], items[0])
+        if not publish_template:
+            widget.publish_to_shotgrid.hide()
+            widget.publish_to_shotgrid.setChecked(False)
+        else:
+            if "publish_to_sg" in ui_settings.keys():
```

Comment:
Suggestion: when you want to check that key exists in a dictionary or iterate over the keys in a dictionary, you can omit the call to `keys()`:
```suggestion
            if "publish_to_sg" in ui_settings:
```